numba/core/lowering.py

```diff
@@ -1309,11 +1309,15 @@ def alloca_lltype(self, name, lltype):
         aptr = cgutils.alloca_once(self.builder, lltype,
                                    name=name, zfill=False)
         if is_uservar:
+            # If it's an arg set the location as the function definition line
+            if name in self.func_ir.arg_names:
+                loc = self.loc.with_lineno(self.func_ir.loc.line)
+            else:
+                loc = self.loc
```

Comment:
I couldn't see anything that tests this. Do you think the following addition is a reasonable test?

```patch
diff --git a/numba/tests/test_debuginfo.py b/numba/tests/test_debuginfo.py
index 7a1aae08f..e9d2fc410 100644
--- a/numba/tests/test_debuginfo.py
+++ b/numba/tests/test_debuginfo.py
@@ -182,6 +182,16 @@ class TestDebugInfoEmission(TestCase):
             dilocation_info = metadata_definition_map[k]
             self.assertIn(f'line: {line_no}', dilocation_info)
 
+        # Check that the line number of the argument `a` is the line number
+        # that the function begins on
+        line_re = re.compile(r'.*!DILocalVariable\(name: "a",.*line: ([0-9]*),')
+        for line in metadata:
+            matched = line_re.match(line)
+            if matched:
+                g = matched.groups()
+                self.assertEqual(len(g), 1)
+                self.assertEqual(int(g[0]), pysrc_line_start)
+
 
 if __name__ == '__main__':
     unittest.main()
```

Reply:
Thanks, good point, I'll add something to this effect.